android/src/main/java/com/oney/WebRTCModule/MediaRecorderImpl.java

```diff
@@ -34,10 +42,16 @@ protected void start(File file) throws Exception {
 
         // Make sure videoTrack exists before adding sink
         if (videoTrack != null) {
-            // Create a new VideoAudioFileRenderer using the file path and EglUtils.getRootEglBaseContext()
-            // VideoAudioFileRenderer will write to the file
-            // Add sink to videoTrack (https://chromium.googlesource.com/external/webrtc/+/HEAD/sdk/android/api/org/webrtc/VideoTrack.java#31)
-            // Log.i(TAG, "Started media recorder! " + videoAudioFileRenderer.toString());
+            // Make sure directories exists before writing a file
+            file.getParentFile().mkdirs();
```

Comment:
This is the only safety check I've added for file safety. I could definitely be more thorough than this if you'd like. I also kinda hate how they did this, but I know it's stolen straight from flutter so it must be effective.

Reply:
Honestly this is up to how much responsibility we want to put on the caller. Also note that in theory whatever we decide impacts the ios as to not make one magically handle folder creations for you and the other doesn't. I think in most cases it's fine for us to do this especially if it doesn't just throw an error if the directories already exist then it's a nicety that's honestly fine.